### PR TITLE
Guardrails policies (M1)

### PR DESCRIPTION
**Pull request by yjst2012**
### Which issue this PR addresses:

Fixes:
https://issues.redhat.com/browse/ARO-2560
https://issues.redhat.com/browse/ARO-2885
https://issues.redhat.com/browse/ARO-2222
https://issues.redhat.com/browse/ARO-2322
https://issues.redhat.com/browse/ARO-2146
https://issues.redhat.com/browse/ARO-1393

### What this PR does / why we need it:

added 4 Guardrails policies:
aro-deny-host-mount
aro-deny-labels
aro-deny-master-toleration-taints
aro-deny-privileged-namespace
under [path](https://github.com/Azure/ARO-RP/tree/feature/guardrails-policy/pkg/operator/controllers/guardrails/policies/gktemplates-src)

### Test plan for issue:

policies have been tested with opa, gator, and on different dev clusters

### Is there any documentation that needs to be updated for this PR?

partially refer to [readme](https://github.com/Azure/ARO-RP/blob/feature/guardrails-policy/pkg/operator/controllers/guardrails/policies/README.md)
Cx facing doc will be addressed separately in another jira https://issues.redhat.com/browse/ARO-3203

### CODE REVIEW

File: pkg/operator/controllers/guardrails/policies/README.md [hunk 1]
```diff
@@ -31,54 +31,53 @@ Constraint is manually created
 apiVersion: templates.gatekeeper.sh/v1
 kind: ConstraintTemplate
 metadata:
-  name: aroprivilegednamespace
+  name: arodenyprivilegednamespace
```
Comment (hlipsig) [inline pkg/operator/controllers/guardrails/policies/README.md:34]: I don't have a lot of context for most of this, but what's the reason for adding deny to the string, and how confusing will that make any other parts of the system/onboarding this change? I assume this naming clarifies functionality, but changing names can be confusing. Is the change documented?

File: pkg/operator/controllers/guardrails/policies/README.md [hunk 2]
```diff
@@ -212,25 +221,54 @@ spec:
           cpu: "100m"
           memory: "30Mi"
 ```
-the assertions section is the expected result
+the `assertions` section is the expected result
 
-gator test is done via cmd:
+gator test can be done via cmd:
```
Comment (hlipsig) [inline pkg/operator/controllers/guardrails/policies/README.md:226]: Should Gator be capitalized here? Noticed you fixed that in a few places but not here.

**Review by hlipsig — COMMENTED** [review]
Questions I have here shouldn't be considered blocking. Just questions from the perspective of being very new to the project.

File: pkg/operator/controllers/guardrails/policies/README.md (hunk 2 above)
Comment (yjst2012) [reply pkg/operator/controllers/guardrails/policies/README.md:226]: Aha, good catch, as [the official website](https://open-policy-agent.github.io/gatekeeper/website/docs/gator/) is using lower case all the time, I'll follow that as well to make it more consistent

File: pkg/operator/controllers/guardrails/policies/README.md (hunk 1 above)
Comment (yjst2012) [reply pkg/operator/controllers/guardrails/policies/README.md:34]: the story is a bit long, originally this policy does not have a deny in its name, and while more and more policies added, all others have arodeny prefix to imply this is a deny policy, and we decided to use that as sort of uniforms, hence is the change.
as to the confusion, the policy is not published yet, this is sort of last chance to unify our namings, the customer facing doc has adapted to the change IIRC.

Comment (hlipsig) [reply pkg/operator/controllers/guardrails/policies/README.md:34]: That makes sense. Thanks.

File: pkg/operator/controllers/guardrails/policies/README.md (hunk 2 above)
Comment (yjst2012) [reply pkg/operator/controllers/guardrails/policies/README.md:226]: change made, thank you for the review

**Review by jaitaiwan — APPROVED** [review]
LGTM!

**Review by hlipsig — APPROVED** [review]
LGTM. But I don't have owner or anything so we still need some one with the powers for merge.

**Review by hawkowl — APPROVED** [review]

**Pull request merged**